akka-http-oauth2-provider/src/main/scala/scalaoauth2/provider/Oauth2Provider.scala

```diff
@@ -0,0 +1,63 @@
+package scalaoauth2.provider
+
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.server.Directives
+import akka.http.scaladsl.server.directives.Credentials
+import scalaoauth2.provider.OAuth2Provider.TokenResponse
+import spray.json.DefaultJsonProtocol
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{ Failure, Success }
+
+trait OAuth2Provider[U] extends Directives with DefaultJsonProtocol {
+  import OAuth2Provider.tokenResponseFormat
+  import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+
+  val oauth2DataHandler: DataHandler[U]
+
+  val tokenEndpoint = new TokenEndpoint {
+    override val handlers = Map(
```

Comment:
All authentications are enabled by default, it is not good to keep security.
Developers will use authentication they do not use without being aware.

Could you change this like [Play subproject](https://github.com/nulab/scala-oauth2-provider/blob/0.17.2/play2-oauth2-provider/src/main/scala/scalaoauth2/provider/OAuth2Provider.scala#L16)?